src/typecheck.c

```diff
@@ -641,7 +642,7 @@ static const Type *check_binop(
     if (got_numbers && !got_integers)
         cast_type = (lhstypes->type == doubleType || rhstypes->type == doubleType) ? doubleType : floatType;
     if (got_pointers)
-        cast_type = voidPtrType;
+        cast_type = longType;
```

Comment:
Could be slightly safer to use unsigned long

```suggestion
        cast_type = get_integer_type(64, false);  // unsigned long
```